js/ui/date_box/ui.date_box.strategy.calendar_with_time.js

```diff
@@ -161,8 +161,30 @@ const CalendarWithTimeStrategy = CalendarStrategy.inherit({
         return this._timeView._hourBox.$element().find('input');
     },
 
-    _attachTabHandler: function() {
+    _toolbarItemCount() {
+        return this.dateBox.option('dropDownOptions.toolbarItems').length;
+    },
+
+    _popupFocusableBoundarySelectors() {
+        if(this._toolbarItemCount() === 0) {
+            return {
+                first: '.dx-item:first-child .dx-texteditor-input',
```

Comment:
I think we should extract it to separate constant variables. At least base classes .dx-texteditor-input, .dx-button to avoid duplication.

Reply:
Fixed in [d926b33defc91a284d9191741af5b9c9f64cf3fd]